### PR TITLE
W-10890540-database-1.14-stored-procedure-named-parameters-kt

### PR DESCRIPTION
**Pull request by kevintroller**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by Prannjall11 — DISMISSED** [review]

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 1]
```diff
@@ -0,0 +1,299 @@
+= Troubleshooting Database Connector - Mule 4
+
+To troubleshoot Anypoint Connector for Database (Database Connector), become familiar with the information about enabling verbose logging, troubleshooting pooling profile connections, troubleshooting Snowflake JDBC driver, and interpreting commonly thrown messages.
+
+Database Connector follows the standards and guidelines detailed by the JDBC library (working with Java SE JDK 1.8).
+
+== Enable Verbose Logging
+
+To get a better understanding of why an application's interaction with the database is failing, temporarily enable verbose logging for Database Connector. +
+Choose one of two options to enable verbose logging: manually change the `log4j2.xml` file or via Anypoint Runtime Manager if you are using CloudHub APIs. +
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]: I would use bullets for the two options because it is easier to scan them.

* Manually change the `log4j2.xml` file.
* Use Anypoint Runtime Manager if you are using CloudHub APIs.

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+Remember to always disable the enhanced verbosity after troubleshooting because it affects your Mule application performance.
+
+To enable verbose logging in the configuration file:
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]: To tie this in better with the first bullet above - 
in the configuration file > in the `log4j2.xml` file

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+. Access Anypoint Studio and navigate to the *Package Explorer* view.
+. Open your application's project name.
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]:
```suggestion
. Open your application's project.
```

Comment (dhirsch1) [reply db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]: You are not opening a name - you are opening a project.

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+			</db:stored-procedure>
+		<logger level="INFO" message="#[output application/json --- payload.resultSet1]" />
+	</flow>
+----
+
+== Understand Select and Stored Procedure Operations Warning Exceptions
+
+When the Database Connector *Select* and *Stored procedure* operations execute correctly but do not return results, Mule runtime engine (Mule) returns a `WARN` exception message instead of throwing the exception.
+
+For the *Select* operation, the behavior occurs when executing an invalid SQL statement, and for the *Store procedure* operation, when there is an error within a set of statements that can collapse the procedure. In fact, JDBC returns a series of `ResultSets`, `UpdateCounts`, and `Exceptions`.
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]:
```suggestion
For the *Select* operation, the behavior occurs when executing an invalid SQL statement. For the*Store procedure* operation, an error within a set of statements can collapse the procedure. In fact, JDBC returns a series of `ResultSets`, `UpdateCounts`, and `Exceptions`.
```

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+Mule does not throw an exception for the flow to catch it because the root cause is in the driver's behavior. Some drivers comply strictly with JDBC and properly throw an exception at execution time, but other drivers do not. Therefore, adding in the connector code the behavior of throwing an exception would break backward compatibility for Mule apps that do not expect an exception.
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc:92]: I can't tell if this is referring to both operations or only to the second one. It might be good to separate the discussion of these operations.

**Review by dhirsch1 — CHANGES_REQUESTED** [review]
I stopped in the middle because I think you should separate the discussion of Select operation from the discussion of the Store procedure operation. It is hard to follow the way it is currently written.

**Review by dhirsch1 — COMMENTED** [review]
I stopped in the middle because I think you should separate the information for the *Select* operation, from the information for the *Store procedure* operation. Having them interspersed with each other makes it very hard to follow.

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc (hunk 5 above)
Comment (kevintroller) [reply db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc:92]: This refers to both operations.

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+		IS
+		BEGIN
+			INSERT INTO TABLE_DEMO_DEFAULT_VALUE (PARAMETER_A, PARAMETER_B, PARAMETER_C)
+			VALUES (PARAMETERA, PARAMETERB, PARAMETERC);
+		END;
+---
+
+=== Scenario: Mixing Named and Ordinal Parameter Bindings
+
+Suppose the following stored procedure execution that uses both named and ordinal parameter bindings:
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]:
```suggestion
Suppose the following stored procedure execution uses both named and ordinal parameter bindings:
```

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+-Djavax.xml.parsers.SAXParserFactory=com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl
+-Djavax.xml.parsers.DocumentBuilderFactory=com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl
+----
+
+These properties avoid any trouble if, for instance, you need to set up Oracle .jars as shared libraries.
+
+For more information about how to set system properties, refer to the xref:mule-runtime::mule-app-properties-system.adoc[system properties] documentation.
+
+
+== Troubleshoot Stored Procedure Operation With Named Parameters
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]:
```suggestion
== Troubleshoot the Stored Procedure Operation With Named Parameters
```

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+
+[source,xml,linenums]
+----
+	Root Exception stack trace:
+	org.mule.runtime.api.exception.MuleRuntimeException: Could not resolve query: call SP_DEMO(PARAMETERB => :parameterb, PARAMETERC=> :parameterc, :parametera).Error: org.mule.db.commons.internal.parser.QueryTemplateParsingException: Invalid Stored Procedure expression: Ordinal binding and Named binding cannot be combined
+----
+
+=== Scenario: JDBC Driver Does Not Support Named Parameters Binding
+
+Suppose the JDBC driver used does not support the named parameters binding feature, the following exception is expected:
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc]:
```suggestion
If the JDBC driver used does not support the named parameters binding feature, the following exception is expected:
```

File: db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+----
+
+The following exception is expected:
+
+[source,xml,linenums]
+----
+	Root Exception stack trace:
+	org.mule.runtime.api.exception.MuleRuntimeException: Could not resolve query: call SP_DEMO(PARAMETERB => :parameterb, PARAMETERC=> :parameterc, :parametera).Error: org.mule.db.commons.internal.parser.QueryTemplateParsingException: Invalid Stored Procedure expression: Ordinal binding and Named binding cannot be combined
+----
+
```
Comment (dhirsch1) [inline db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc:231]: I would remove "Scenario:" here and elsewhere. I don't think it's necessary and it makes things harder to skim.

**Review by dhirsch1 — DISMISSED** [review]

**Comment by kevintroller** [comment]
Closing as this version was not released.
